### PR TITLE
checksec wrapper

### PR DESCRIPTION
**Pull request by crmulliner**
add a wrapper for checksec

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ Always update Version in Makefile
 ## Unreleased
 
 ### Added
+- checksec wrapper script, see [check_sec.sh](scripts/check_sec.sh) and Checksec.md
```
Comment (Grazfather) [inline Changelog.md:9]: let's link to the markdown as well

File: Checksec.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
+# checksec Integration
+
+[checksec](https://github.com/slimm609/checksec.sh) is a bash script for checking security properties of executables (like PIE, RELRO, Canaries, ...).
+
+Checksec is an incredible helpful tool therefore we developed a wrapper script for FwAnalyzer to ease the usage of checksec. Below
+we go thru the steps required to use checksec with FwAnalyzer.
```
Comment (Grazfather) [inline Checksec.md:6]: thru -> through

File: Checksec.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+Configuration is done in two steps. First step is adding a `FileContent` check that uses the `Script` option.
+The second step is creating the checksec wrapper configuration. The configuration allows you to selectively skip files
+(e.g. vendor binaries) and fine tune the security features that you want to enforce.
+
+### checksec wrapper configuration
+
+The checksec wrapper has two options, and uses JSON:
+
+- cfg : checksec config, where you can select acceptable values for each field in the checksec output. The `key` is the name of the checksec field and the value is an array where each item is an acceptable value (e.g. allow `full` and `partial` RELRO). Omitted fields are not checked.
```
Comment (Grazfather) [inline Checksec.md:23]: why is key in backticks

File: scripts/check_sec.sh [hunk 4]
```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+FILEPATH=$1
+ORIG_FILENAME=$2
+ORIG_UID=$3
+ORIG_GID=$4
+ORIG_MODE=$5
+ORIG_LABEL=$6
+CONFIG=$8
```
Comment (Grazfather) [inline scripts/check_sec.sh:9]: $7?

Comment (crmulliner) [reply scripts/check_sec.sh:9]: is always "--"

File: scripts/check_sec.sh [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+      passed = False
+      for expected_value in expected["cfg"][k]:
+        if expected_value == result[fp][k]:
+          passed = True
+          break
+      if not passed:
+        print(json.dumps(result[fp]).rstrip())
+        sys.exit(0)
+except Exception as e:
+  #print(e)
```
Comment (Grazfather) [inline scripts/check_sec.sh:70]: remove?

File: scripts/check_sec.sh [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+fp = os.getenv("FILEPATH")
+orig_name = os.getenv("ORIG_FILENAME")
+
+try:
+  expected = json.loads(cfg.rstrip())
+  result = json.loads(res.rstrip())
+ 
+  if "skip" in expected: 
+    if orig_name in expected["skip"]:
+      #print("skipping " + orig_name)
```
Comment (Grazfather) [inline scripts/check_sec.sh:56]: remove?

File: scripts/check_sec.sh [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+      sys.exit(0)
+
+  for k in expected["cfg"]:
+    if k in result[fp]:
```
Comment (Grazfather) [inline scripts/check_sec.sh:60]: could maybe flip to avoid nesting: `if k not in result[fp]: continue`

**Pull request merged**